web/pdf_sidebar_resizer.js

```diff
@@ -0,0 +1,149 @@
+/* Copyright 2017 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NullL10n } from './ui_utils';
+
+const SIDEBAR_WIDTH_VAR = '--sidebar-width';
+const SIDEBAR_MIN_WIDTH = 200; // pixels
+const SIDEBAR_RESIZING_CLASS = 'sidebarResizing';
+
+/**
+ * @typedef {Object} PDFSidebarResizerOptions
+ * @property {HTMLDivElement} outerContainer - The outer container
+ *   (encasing both the viewer and sidebar elements).
+ * @property {HTMLDivElement} sidebarContainer - The sidebar container
+ *   (in which the sidebar view element are placed).
+ * @property {HTMLDivElement} resizer - The DOM element that can be dragged in
+ *   order to adjust the width of the sidebar.
+ */
+
+class PDFSidebarResizer {
+  /**
+   * @param {PDFSidebarResizerOptions} options
+   * @param {EventBus} eventBus - The application event bus.
+   * @param {IL10n} l10n - Localization service.
+   */
+  constructor(options, eventBus, l10n = NullL10n) {
+    this.enabled = false;
+    this.isRTL = false;
+    this.doc = document.documentElement;
+    this._boundEvents = Object.create(null);
+
+    this.outerContainer = options.outerContainer;
+    this.sidebarContainer = options.sidebarContainer;
+    this.resizer = options.resizer;
+    this.eventBus = eventBus;
+    this.l10n = l10n;
+
+    if (typeof CSS === 'undefined' || typeof CSS.supports !== 'function' ||
+        !CSS.supports(SIDEBAR_WIDTH_VAR, `calc(-1 * ${SIDEBAR_MIN_WIDTH}px)`)) {
+      console.warn('PDFSidebarResizer: ' +
+                   'The browser does not support resizing of the sidebar.');
+      this.resizer.classList.add('hidden'); // Hide the resizer DOM element.
+      return;
+    }
+    this.enabled = true;
+    this.resizer.classList.remove('hidden'); // Show the resizer DOM element.
+
+    this.l10n.getDirection().then((dir) => {
+      this.isRTL = (dir === 'rtl');
+    });
+    this._addEventListeners();
+  }
+
+  /**
+   * @private
+   */
+  _updateWidth(width = 0) {
+    if (!this.enabled) {
+      return;
+    }
+    // Prevent the sidebar from becoming too narrow, or from occupying more
+    // than half of the available viewer width.
+    const maxWidth = Math.floor(this.outerContainer.clientWidth / 2);
```

Comment:
Can't we precalculate this in the constructor so we don't have to on every call to `_updateWidth`?

Reply:
> Can't we precalculate this so we don't have to on every call to _updateWidth?

Sure, that is a really nice idea!
I've added a getter for this value, and I don't think it's making the code unnecessarily complex.